library/src/main/java/com/cube/storm/ui/quiz/view/holder/ImageQuizItemViewHolder.java

```diff
@@ -98,50 +104,82 @@ else if (index >= images.size())
 		}
 	}
 
+	/**
+	 * Handles the view changes when an image answer is selected or unselected
+	 * @param answerLayout the layout container other views
+	 * @param selectAnswer if answer is selected or unselected
+	 */
+	private void selectAnswer(View answerLayout, boolean selectAnswer)
+	{
+		// Set border around image
+		View border = answerLayout.findViewById(R.id.image_border);
+		if (border != null)
+		{
+			border.setVisibility(selectAnswer ? View.VISIBLE : View.INVISIBLE);
+		}
+
+		// Set image label style and its background
+		TextView textView = answerLayout.findViewById(R.id.label);
+		if (textView != null)
+		{
+			textView.setTextAppearance(answerLayout.getContext(),
+				selectAnswer ? R.style.QuizImageLabel_Selected :R.style.QuizImageLabel_Unselected);
+			textView.setBackgroundResource(selectAnswer ? R.drawable.quiz_image_label_selected_border : 0);
+		}
+	}
+
 	private class ModelClickListener implements OnClickListener
 	{
 		private ImageQuizItem model;
+		private int index;
 
-		private ModelClickListener(ImageQuizItem model)
+		private ModelClickListener(ImageQuizItem model, int index)
 		{
 			this.model = model;
+			this.index = index;
 		}
 
 		@Override public void onClick(View v)
 		{
-			int index = (Integer)v.getTag(R.id.checkbox);
-			CheckBox checker = ((CheckBox)v.findViewById(R.id.checkbox));
-
-			if (!checker.isChecked())
+			if (answerLayouts != null && answerLayouts.get(index) != null)
 			{
-				checker.setChecked(true);
-				model.getSelectHistory().add(index);
+				View answerLayout = answerLayouts.get(index);
 
-				for (QuizEventHook quizEventHook : QuizSettings.getInstance().getEventHooks())
+				if (!model.getSelectHistory().contains(index))
 				{
-					quizEventHook.onQuizOptionSelected(v.getContext(), itemView, model, index);
+					// select image answer
+					for (QuizEventHook quizEventHook : QuizSettings.getInstance().getEventHooks())
+					{
+						quizEventHook.onQuizOptionSelected(v.getContext(), itemView, model, index);
+					}
+					selectAnswer(answerLayout, true);
+					model.getSelectHistory().add(index);
+				}
+				else
+				{
+					// unselect image answer
+					selectAnswer(answerLayout, false);
+					model.getSelectHistory().remove((Integer)index);
 				}
-			}
-			else
-			{
-				checker.setChecked(false);
-				model.getSelectHistory().remove((Integer)index);
 			}
 
-			// disable select if checked > limit
+			// unselect first selected answer if selected answers goes over answer limit
 			if (model.getSelectHistory().size() > model.getLimit())
 			{
 				int remIndex = model.getSelectHistory().get(0);
 				model.getSelectHistory().remove(0);
+				View answerLayout;
 
 				if (remIndex % 2 == 0)
 				{
-					((CheckBox)options.getChildAt((int)Math.floor(remIndex / 2)).findViewById(R.id.layout1).findViewById(R.id.checkbox)).setChecked(false);
+					answerLayout = options.getChildAt((int) Math.floor(remIndex / 2)).findViewById(R.id.layout1);
```

Comment:
There's two different ways of getting an `answerLayout` in this method. One is using `options.getChildAt`, the other is using `answerLayouts.get`. Could you switch them to be consistent?

Reply:
Good point! Made a code a whole lot more readable now.